### ISSUE TITLE
createFunction returns incorrect result from dunction

### ISSUE THREAD

**Issue by oceanwap** [sqlite/sqlite-wasm#59]
Sample code
https://stackblitz.com/edit/vitejs-vite-bwgxcr?file=index.html

Expected Sum: 5
Returned Sum: 718664

**Comment by tomayac** [comment]
@oceanwap, please follow the [SQLite bug filing instructions](https://www.sqlite.org/src/wiki?name=Bug+Reports). 

@sgbeal FYI.

**Comment by sgbeal** [comment]
Cannot reproduce. On my system the linked-to demo produces the proper output:

Firefox v121:

```
SQL TRACE #1 via sqlite3@657088: select sum_a(?, ?) as sum [sqlite3-bundler-friendly.mjs:10432:23](https://vitejsvitebwgxcr-wmza--5173--fe7beb95.local-corp.webcontainer.io/node_modules/@sqlite.org/sqlite-wasm/sqlite-wasm/jswasm/sqlite3-bundler-friendly.mjs?v=e234a3b6)
5
```

Chrome v120:

```
SQL TRACE #1 via sqlite3@657088: select sum_a(?, ?) as sum
main.js:7 5
```

Also: it's running version 3.42, which is ancient by the standards of the wasm build.

**Comment by tomayac** [comment]
(Just published `v3.45.0-build1` if you want to try with the latest build.)
